File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    adict = dict()
    for i in range(len(clothes)) :
        if clothes[i][1] in adict:
            adict[clothes[i][1]] += 1
        else :
            adict[clothes[i][1]] = 1
    
    print(adict)
    for value in adict.values() :
        answer *= (value+1)
    
    return answer-1

File: 프로그래머스/lv2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 1;
    let amap = new Map();
    
    for(let c of clothes) {
        if(!amap.has(c[1])) amap.set(c[1], 0);
        if(amap.has(c[1])) amap.set(c[1], amap.get(c[1]) + 1);
    }
    for(let [k, v] of amap) {
        answer *= (v+1);
    }
    
    return answer-1;
}